sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -800,6 +860,58 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             }
         }
 
+        private async Task<ServiceBusReceiver> CreateReceiverIfNotExist(
+            string sessionId,
+            ServiceBusReceiverOptions receiverOptions,
+            CancellationToken cancellationToken)
+        {
+            ServiceBusReceiver receiver = null;
+            await receiverCreationLockMap[sessionId].WaitAsync(cancellationToken).ConfigureAwait(false);
+            try
+            {
+                if (!sessionIdsReceiverMap.ContainsKey(sessionId))
+                {
+                    receiver = await CreateReceiver(receiverOptions, cancellationToken, sessionId).ConfigureAwait(false);
+                }
+                else
+                {
+                    receiver = sessionIdsReceiverMap[sessionId];
+                }
+            }
+            finally
+            {
+                receiverCreationLockMap[sessionId].Release();
+            }
+
+            return receiver;
+        }
+
+        private async Task<ServiceBusReceiver> CreateReceiver(
```

Comment:
This is only for session receivers. Should it return a ServiceBusSessionReceiver and rename to CreateAndInitializeSessionReceiver?